src/coreclr/jit/optimizemaskconversions.cpp

```diff
@@ -147,19 +153,48 @@ class MaskConversionsCheckVisitor final : public GenTreeVisitor<MaskConversionsC
                     hasConversion = true;
                 }
                 break;
+            }
 
             case GT_LCL_VAR:
+            {
                 isLocalUse = true;
 
                 // Look for:
-                //      user:ConvertVectorToMask(use:LCL_VAR(x)))
+                //      user: ConvertVectorToMask(use:LCL_VAR(x)))
+                // -or-
+                //      user: ConditionalSelect(use:LCL_VAR(x), y, z)
 
-                if (user->OperIsConvertVectorToMask())
+                if (user->OperIsHWIntrinsic())
                 {
-                    convertOp     = user->AsHWIntrinsic();
-                    hasConversion = true;
+                    GenTreeHWIntrinsic* hwintrin = user->AsHWIntrinsic();
+                    NamedIntrinsic      ni       = hwintrin->GetHWIntrinsicId();
+
+                    if (hwintrin->OperIsConvertVectorToMask())
+                    {
+                        convertOp     = user->AsHWIntrinsic();
+                        hasConversion = true;
+                    }
+                    else if (hwintrin->OperIsVectorConditionalSelect())
+                    {
+                        // We don't actually have a convert here, but we do have a case where
+                        // the mask is being used in a ConditionalSelect and therefore can be
+                        // consumed directly as a mask. While the IR shows TYP_SIMD, it gets
+                        // handled in lowering as part of the general embedded-mask support.
```

Comment:
In other words the conditional select operations support both `TYP_SIMD` and `TYP_MASK` for the mask operand, right? With the `TYP_SIMD` one being a 0/1 in each lane, and the `TYP_MASK` one being a bit mask.

Reply:
> In other words the conditional select operations support both TYP_SIMD and TYP_MASK for the mask operand, right?

Right.

> With the TYP_SIMD one being a 0/1 in each lane, and the TYP_MASK one being a bit mask.

Rather `TYP_SIMD` is `Zero` or `AllBitsSet` in each lane and `TYP_MASK` is a compressed form being a bitmask (1-bit per element).

This comes about from SIMD comparisons returning `Zero`/`AllBitsSet` per lane such that it can be used with all bitmask operations, not just with conditional select or similar.